packages/core/src/external/commonwell/document/document-file-type-detector.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * The function `detectFileType` uses magic numbers to determine the file type of a given file.
+ * Magic numbers are unique sequences of bytes that identify the file format or protocol.
+ * In this case, the function reads the first 5 bytes (the magic number) of the file buffer and
+ * matches it against known file type headers.
+ *
+ * @param {Buffer} fileBuffer - The `fileBuffer` parameter is a `Buffer` object that represents the
+ * contents of a file. The first 5 bytes of this buffer, which form the magic number, are used to
+ * identify the file type.
+ * @returns The function `detectFileType` returns a string representing the detected file type.
+ */
+export function detectFileType(fileBuffer: Buffer): string {
```

Comment:
any logic to make this more efficient? probably not nessecary given that this is an edge case anyway